metaflow/plugins/aws/batch/batch.py

```diff
@@ -161,8 +154,49 @@ def launch_job(
         if attrs:
             for key, value in attrs.items():
                 job.parameter(key, value)
-        self.job = job.execute()
+        return job
 
+    def launch_job(
+        self,
+        step_name,
+        step_cli,
+        code_package_sha,
+        code_package_url,
+        code_package_ds,
+        image,
+        queue,
+        iam_role=None,
+        cpu=None,
+        gpu=None,
+        memory=None,
+        run_time_limit=None,
+        env={},
+        attrs={}
+    ):
+        if queue is None:
+            queue = next(self._client.active_job_queues(), None)
+            if queue is None:
+                raise BatchException(
+                    'Unable to launch Batch job. No job queue '
+                    ' specified and no valid & enabled queue found.'
+                )
+        job = self.create_job(
```

Comment:
Interesting - what is the difference between the types returned by create_job and job.execute()?

Reply:
`job.execute` returns a reference to the running job on AWS Batch. `create_job` only creates the spec for the job to be executed.